ergotree-interpreter/src/eval/bin_op.rs

```diff
@@ -174,26 +174,33 @@ impl Evaluable for BinOp {
     fn eval(&self, env: &Env, ctx: &mut EvalContext) -> Result<Value, EvalError> {
         ctx.cost_accum.add(Costs::DEFAULT.eq_const_size)?;
         let lv = self.left.eval(env, ctx)?;
-        // using closure to keep right value from evaluation (for lazy AND, OR)
+        // using closure to keep right value from evaluation (for lazy AND, OR, XOR)
         let mut rv = || self.right.eval(env, ctx);
         match self.kind {
-            BinOpKind::Relation(op) => match op {
-                RelationOp::Eq => Ok(Value::Boolean(lv == rv()?)),
-                RelationOp::NEq => Ok(Value::Boolean(lv != rv()?)),
-                RelationOp::Gt => eval_gt(lv, rv()?),
-                RelationOp::Lt => eval_lt(lv, rv()?),
-                RelationOp::Ge => eval_ge(lv, rv()?),
-                RelationOp::Le => eval_le(lv, rv()?),
-                RelationOp::And => Ok(Value::Boolean(if lv.try_extract_into::<bool>()? {
+            BinOpKind::Logical(op) => match op {
+                LogicalOp::And => Ok(Value::Boolean(if lv.try_extract_into::<bool>()? {
                     rv()?.try_extract_into::<bool>()?
                 } else {
                     false
                 })),
-                RelationOp::Or => Ok(Value::Boolean(if !lv.try_extract_into::<bool>()? {
+                LogicalOp::Or => Ok(Value::Boolean(if !lv.try_extract_into::<bool>()? {
                     rv()?.try_extract_into::<bool>()?
                 } else {
                     true
                 })),
+                LogicalOp::Xor => Ok(Value::Boolean(if lv.try_extract_into::<bool>()? {
```

Comment:
I meant to use rust ^ XOR in op eval as well.

Reply:
Fixed